internal/utils/gke.go

```diff
@@ -0,0 +1,508 @@
+			Network:    *config.Spec.NetworkConfig.Network,
+		}
+	}
+
+	if config.Spec.NetworkPolicy != nil {
+		request.Cluster.NetworkPolicy = &gkeapi.NetworkPolicy{
+			Enabled:  *config.Spec.NetworkPolicy.Enabled,
+			Provider: *config.Spec.NetworkPolicy.Provider,
+		}
+	}
+
+	if config.Spec.PrivateClusterConfig != nil {
+		request.Cluster.PrivateClusterConfig = &gkeapi.PrivateClusterConfig{
+			EnablePrivateEndpoint: *config.Spec.PrivateClusterConfig.EnablePrivateEndpoint,
+			EnablePrivateNodes:    *config.Spec.PrivateClusterConfig.EnablePrivateNodes,
+			MasterIpv4CidrBlock:   config.Spec.PrivateClusterConfig.MasterIpv4CidrBlock,
+			PrivateEndpoint:       config.Spec.PrivateClusterConfig.PrivateEndpoint,
+			PublicEndpoint:        config.Spec.PrivateClusterConfig.PublicEndpoint,
+		}
+	}
+
+	return request, nil
+}
+
+// WaitClusterRemoveExp waits for a cluster to be removed
+func WaitClusterRemoveExp(ctx context.Context, svc *gkeapi.Service, config *gkev1.GKEClusterConfig) (*gkeapi.Operation, error) {
+	var operation *gkeapi.Operation
+	var err error
+
+	for i := 1; i < 12; i++ {
+		time.Sleep(time.Duration(i*i) * time.Second)
+		operation, err = svc.Projects.Locations.Clusters.Delete(ClusterRRN(config.Spec.ProjectID, config.Spec.Region, config.Spec.ClusterName)).Context(ctx).Do()
+		if err == nil {
+			return operation, nil
+		} else if !strings.Contains(err.Error(), "Please wait and try again once it is done") {
+			break
+		}
+	}
+	return operation, err
+}
+
+// GetClientset returns the clientset for a cluster
+func GetClientset(cluster *gkeapi.Cluster, ts oauth2.TokenSource) (kubernetes.Interface, error) {
+	capem, err := base64.StdEncoding.DecodeString(cluster.MasterAuth.ClusterCaCertificate)
+	if err != nil {
+		return nil, err
+	}
+	host := cluster.Endpoint
+	if !strings.HasPrefix(host, "https://") {
+		host = fmt.Sprintf("https://%s", host)
+	}
+
+	config := &rest.Config{
+		Host: host,
+		TLSClientConfig: rest.TLSClientConfig{
+			CAData: capem,
+		},
+		WrapTransport: func(rt http.RoundTripper) http.RoundTripper {
+			return &oauth2.Transport{
+				Source: ts,
+				Base:   rt,
+			}
+		},
+	}
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+
+	return clientset, nil
+}
+
+// GetGKEClient returns a gke client capable of making requests on behalf of the services account
+func GetGKEClient(ctx context.Context, secretsCache wranglerv1.SecretCache, config *gkev1.GKEClusterConfig) (*gkeapi.Service, error) {
+	ns, id := Parse(config.Spec.CredentialContent)
+	secret, err := secretsCache.Get(ns, id)
+
+	if err != nil {
+		return nil, err
+	}
+
+	dataBytes := secret.Data["gkeCredentialConfig-data"]
+
+	data := string(dataBytes)
+
+	return GetServiceClient(ctx, data)
+}
+
+// ValidateCreateRequest checks a config for the ability to generate a create request
+func ValidateCreateRequest(config *gkev1.GKEClusterConfig) error {
+	//TODO: check if these can even be nill/empty in a inported cluster.
```

Comment:
Should this be removed or has it not been investigated yet?

Reply:
Removed